### PR TITLE
SUR-232 Implement Pagination Component in Force UI as per Figma - Fixed tab accessibility issues

### PR DESCRIPTION
**Pull request by JelenaTakac**
### Description

Video: https://bsf.d.pr/v/LHbESt

### Checklist:

-   [ ] My code is tested
-   [ ] My code passes the PHPCS tests
-   [ ] I've created the npm build.
-   [ ] My code follows accessibility standards 
-   [ ] My code has proper inline documentation 
-   [ ] I've included any necessary tests 
-   [ ] I've included developer documentation 
-   [ ] I've added proper labels to this pull request

### CODE REVIEW

**Review by jaieds — CHANGES_REQUESTED** [review]
Expected: Disabled pagination items should not be clickable.
Current: Disabled pagination items are clickable and after clicking it opens the address given in the href.

**Review by jaieds — CHANGES_REQUESTED** [review]
Don't change the tag to `button`. Instead, you can remove the href attribute and attach an `onClick` event handler to the link and from that handler just `return false` will resolve the issue.

File: src/components/pagination/pagination.jsx [hunk 1]
```diff
@@ -148,18 +143,16 @@ PaginationNext.displayName = 'Pagination.Next';
...
 			<span
 				className={ cn(
+					'flex justify-center',
 					sizeClassNames[ size ].ellipse,
 					disabled && disabledClassNames.general
 				) }
 				{ ...props }
 			>
-				...
+				•••
```
Comment (BabuTechBsf) [inline src/components/pagination/pagination.jsx:155]: Please use a lucide Icon.

File: src/components/pagination/component-style.js [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
+		icon: 'size-5',
 	},
 	lg: {
-		general: 'px-5 text-base',
-		ellipse: 'px-4 text-base',
-		icon: 'px-3 text-base',
+		general: 'text-base min-w-12 h-12',
+		ellipse: 'text-base min-w-12',
+		icon: 'size-6',
 	},
```
Comment (BabuTechBsf) [inline src/components/pagination/component-style.js:21]: We should not fix heights. Achieve the trequired height by providing the necessary paddings and inner content styles.

File: src/components/pagination/pagination.jsx [hunk 3]
```diff
@@ -78,9 +76,12 @@ const PaginationButton = ( {
 } ) => {
 	const { size, disabled } = usePageContext();
 
+	// Set the tag to 'button' when disabled to avoid clickable links
+	const Tag = disabled ? 'button' : tag;
+
 	return (
 		<Button
-			tag={ tag }
+			tag={ Tag }
```
Comment (BabuTechBsf) [inline src/components/pagination/pagination.jsx:84]: No need of this casesensitive aproach, you can directly use like 
tag={ disabled ? 'button' : tag }

**Review by BabuTechBsf — CHANGES_REQUESTED** [review]
Pagination
Unable to disable individual Item
Target control can be select with options instead of input.
Icon control can be ReactNode
all args details are not available.
https://d.pr/i/LFXk2a - in this case padding looks less. can we make it px-2.
Prepare some informative examples. 
With text navigations, with word page names etc… check breadcrumb examples online.

**Review by ravindra114 — CHANGES_REQUESTED** [review]
https://d.pr/v/VtMegx
onClick/isActive does not look as expected
Can you have a look?

File: src/components/pagination/pagination.jsx (hunk 1 above)
Comment (JelenaTakac) [reply src/components/pagination/pagination.jsx:155]: Akbar suggested that I use three dots like this.

File: src/components/pagination/component-style.js (hunk 2 above)
Comment (JelenaTakac) [reply src/components/pagination/component-style.js:21]: This is UAT feedback from Akbar:
Size:
xs: min-w-6 (24px), h-6 (24px)
sm: min-w-8 (32px), h-8 (32px)
md: min-w-10 (40px), h-10 (40px)
lg: min-w-12 (48px), h-12 (48px)

**Comment by JelenaTakac** [comment]
> Pagination Unable to disable individual Item Target control can be select with options instead of input. Icon control can be ReactNode & one Icon is given for next and right, i guess we need nexticon, prevIcon like this. How to pass next/previous text. all args details are not available. https://d.pr/i/LFXk2a - in this case padding looks less. can we make it px-2. Prepare some informative examples. With text navigations, with word page names etc… check breadcrumb examples online.

I created Jira story for Pagination imporvements: https://brainstormforce.atlassian.net/browse/SUR-330

**Review by BabuTechBsf — APPROVED** [review]

**Review by jaieds — APPROVED** [review]
It is working fine.

https://d.pr/v/e8tDiQ

**Pull request merged**